Fix highlighting of SignColumn and cursor line number

rkiyanchuk opened the pull request:
This pull requests addresses issues with highlighting the columns left of line numbering as well as cursor line number itself.

The `SignColumn` problems has been discovered in issue altercation/solarized#252.

Commits fixing that are authored by @jwhitley and found in his [forked repository](https://github.com/jwhitley/vim-colors-solarized/commits/master).

`CursorLineNr` highlighting problem has been discovered in issue #82.

For easy comparison, this is how highlighting would look like with current upstream solarized:
![upstream solarized](http://s21.postimg.org/9yja9r8tz/2015_03_10_013547_1920x1080_scrot.png)

This is how highlighting looks after the fixes:
![fixed solarized](http://s27.postimg.org/57fqgwwar/2015_03_10_013527_1920x1080_scrot.png)

I'm not pro in vim scheming, so please let me know if further adjustments are needed, I will be glad to improve the pull request.

jwhitley committed:
[vim] Style the SignColumn

`SignColumn` is styled with base03 background

jwhitley committed:
[vim] Add highlight definitions for vim-gitgutter plugin

Commit:
[vim] Fix cursor line number highlight

Fixes issue #82.

Commit:
Remove trailing spaces

Commit:
Add Syntastic highlighting

ELLIOTTCABLE commented:
Is there a reason this was never merged? Is this still pending, or …

rkiyanchuk commented:
@ELLIOTTCABLE, it appears @altercation has been having time scarcity lately. He however told that the project is not going to get abandoned and he'll take care of the pull requests closer to autumn. Until then I think we can just make a fork and merge some of the fixes and improvements, so @altercation can merge those back eventually, making life easier for him.

andyleejordan commented on colors/solarized.vim:
```diff
@@ -654,7 +654,7 @@ exe "hi! DiffDelete"     .s:fmt_none   .s:fg_red    .s:bg_base02
     endif
 endif
-exe "hi! SignColumn"     .s:fmt_none   .s:fg_base0
+exe "hi! SignColumn"     .s:fmt_none   .s:fg_base0  .s:bg_base03
```
I think this should be `s:bg_back` instead so it respects settings that change the background base color.

Commit:
Fix background color

Thanks @andschwa for the hint:

> I think this should be `s:bg_back` instead so it respects settings that change the background base color.

fyodoriv commented:
Using @jwhitley's fork helped me with neomake background issues as well.
